regression_tests/operator.go

```diff
@@ -0,0 +1,262 @@
+	adminXPub           = "xpub661MyMwAqRbcFgfmdkPgE2m5UjHXu9dj124DbaGLSjaqVESTWfCD4VuNmEbVPkbYLCkykwVZvmA8Pbf8884TQr1FgdG2nPoHR8aB36YdDQh"
+	leaderPaymailAlias  = "leader"
+	minimalBalance      = 100
+	defaultGoClientPath = "../../spv-wallet-go-client/regression_tests"
+	defaultJSClientPath = "../../spv-wallet-js-client/regression_tests"
+)
+
+func main() {
+	loadConfigFlag := flag.Bool("l", false, "Load configuration from .env.config file")
+	flag.Parse()
+
+	var config *regressionTestConfig
+	var err error
+	user := &regressionTestUser{}
+
+	if *loadConfigFlag {
+		config, err = loadConfig()
+		if err != nil {
+			fmt.Println("error loading config:", err)
+			return
+		}
+		user.XPriv = config.ClientOneLeaderXPriv
+	} else {
+		config = &regressionTestConfig{}
+	}
+
+	if !isSPVWalletRunning(domainLocalHost) {
+		fmt.Println("spv-wallet is not running. Run spv-wallet and try again")
+		return
+	}
+
+	sharedConfig, err := getSharedConfig(adminXPub)
+	if err != nil {
+		fmt.Println("error getting shared config:", err)
+		return
+	}
+
+	setConfigClientsUrls(config, sharedConfig.PaymailDomains[0])
+
+	if !isSPVWalletRunning(config.ClientOneURL) {
+		fmt.Println("can't connect to spv-wallet at", config.ClientOneURL, "\nEnable tunneling from localhost")
+		return
+	}
+
+	user, err = handleUserCreation(leaderPaymailAlias, config)
+	if err != nil {
+		fmt.Println("error handling user creation:", err)
+		return
+	}
+
+	setConfigLeaderXPriv(config, user.XPriv)
+
+	if err := handleFundsTransfer(user, config); err != nil {
+		fmt.Println("error handling transactions:", err)
+		return
+	}
+
+	clientType := ""
+	for clientType != "go" && clientType != "js" {
+		clientType = promptUser("Do you want to run tests from go-client or js-client? (enter 'go' or 'js'): ")
+		clientType = strings.ToLower(clientType)
+	}
+
+	var defaultPath string
+	if clientType == "go" {
+		defaultPath = defaultGoClientPath
+	} else {
+		defaultPath = defaultJSClientPath
+	}
+	err = saveConfig(config)
+	if err != nil {
+		fmt.Println("error saving config:", err)
+		return
+	}
+	setEnvVariables(config)
+
+	if err := runTests(clientType, defaultPath); err != nil {
+		fmt.Println("error running tests:", err)
+	}
+}
+
+// handleUserCreation handles the creation of a user.
+func handleUserCreation(paymailAlias string, config *regressionTestConfig) (*regressionTestUser, error) {
+	if config.ClientOneLeaderXPriv != "" {
+		if promptUserAndCheck("Would you like to use user from env? (y/yes or n/no): ") == 1 {
+			return useUserFromEnv(config, paymailAlias)
+		}
+	}
+
+	user, err := createUser(paymailAlias, config)
+	if err != nil {
+		return handleCreateUserError(err, paymailAlias, config)
+	}
+
+	return user, nil
+}
+
+// handleCreateUserError handles the error when creating a user.
+func handleCreateUserError(err error, paymailAlias string, config *regressionTestConfig) (*regressionTestUser, error) {
+	if err.Error() == "paymail already exists" {
```

Comment:
Maybe you could define errors once and then check for the specific error using `errors.Is`. 

That approach would prevent from undefined behavior when you change the actual string for some error.

Reply:
After DM conversation - we'll maybe do it in the second iteration

Reply:
Changed to unresolved as more changes were needed. 
For now only predefinition of error was made (errors.Is will be introduced after fixes of errors will be merged).